crates/modelardb_server/src/query/grid_exec.rs

```diff
@@ -373,3 +391,77 @@ impl RecordBatchStream for GridStream {
         self.schema.clone()
     }
 }
+
+/// Metrics collected by [`GridStream`] for use by EXPLAIN ANALYZE.
+#[derive(Debug)]
+struct GridStreamMetrics {
+    /// Default set of metrics collected by operators.
+    baseline_metrics: BaselineMetrics,
+    /// Number of data points reconstructed from segments.
+    created_rows: Count,
+    /// Number of data points reconstructed from segments grouped by model type.
+    created_rows_by_model_type: [Count; 3],
+    /// Number of segments with residuals.
+    segments_with_residuals: Count,
+    /// Number of segments grouped by model type.
+    segments_with_model_type: [Count; 3],
+    /// Number of regular segments.
+    segments_regular: Count,
+    /// Number of irregular segments.
+    segments_irregular: Count,
+}
+
+impl GridStreamMetrics {
+    fn new(metrics: &ExecutionPlanMetricsSet, partition: usize) -> Self {
+        let baseline_metrics = BaselineMetrics::new(metrics, partition);
+
+        let created_rows = Self::new_counter(metrics, partition, "created_rows");
+        let created_rows_by_model_type = [
+            Self::new_counter(metrics, partition, "created_rows_by_pmc_mean"),
+            Self::new_counter(metrics, partition, "created_rows_by_swing"),
+            Self::new_counter(metrics, partition, "created_rows_by_gorilla"),
```

Comment:
OK, I now see that one also has to update this if a new model type is added (unless it just would default to `"created_rows_by_model_type_4"` if not specified. But that is probably not worth the effort...

Could there be a constant array where names of model types have to be specified? The length of that could then be used above and the names here? Maybe an overkill... :-)

Reply:
Fixed by adding `MODEL_TYPE_NAME` which is length `MODEL_TYPE_COUNT` so it should always be the same length as the arrays in `GrimStreamMetrics`. Thus, the metrics should automatically work with new model types if these two constants are updated in the `modelardb_compression` crate which also contains the implementation of all of the model types.